### PR TITLE
chore: wrong namespace

### PR DESCRIPTION
**Pull request by soyuka**

### CODE REVIEW

File: src/State/Provider/DeserializeProvider.php [hunk 1]
```diff
@@ -51,10 +51,7 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             return $data;
         }
 
-        if (
-            !($operation->canDeserialize() ?? true)
-            || !\in_array($method = $operation->getMethod(), ['POST', 'PUT', 'PATCH'], true)
-        ) {
+        if (!$operation->canDeserialize()) {
```
Comment (soyuka) [inline src/State/Provider/DeserializeProvider.php:54]: this is done in the MainController now

File: src/Symfony/Bundle/Resources/config/state.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
             <argument type="service" id="api_platform.serializer.context_builder" />
         </service>
 
-        <service id="api_platform.state_provider.deserialize" class="ApiPlatform\Symfony\State\DeserializeProvider" decorates="api_platform.state_provider.main" decoration-priority="300">
+        <service id="api_platform.state_provider.deserialize" class="ApiPlatform\State\DeserializeProvider" decorates="api_platform.state_provider.main" decoration-priority="300">
```
Comment (vincentchalamon) [inline src/Symfony/Bundle/Resources/config/state.xml:30]:
```suggestion
        <service id="api_platform.state_provider.deserialize" class="ApiPlatform\State\Provider\DeserializeProvider" decorates="api_platform.state_provider.main" decoration-priority="300">
```

File: src/Symfony/Bundle/Resources/config/state.xml [hunk 3]
```diff
@@ -45,7 +45,7 @@
             <argument type="service" id="api_platform.state_processor.add_link_header.inner" />
         </service>
 
-        <service id="api_platform.state_processor.serialize" class="ApiPlatform\Symfony\State\SerializeProcessor" decorates="api_platform.state_processor.main" decoration-priority="200">
+        <service id="api_platform.state_processor.serialize" class="ApiPlatform\State\SerializeProcessor" decorates="api_platform.state_processor.main" decoration-priority="200">
```
Comment (vincentchalamon) [inline src/Symfony/Bundle/Resources/config/state.xml:48]:
```suggestion
        <service id="api_platform.state_processor.serialize" class="ApiPlatform\State\Processor\SerializeProcessor" decorates="api_platform.state_processor.main" decoration-priority="200">
```

File: src/State/Provider/DeserializeProvider.php [hunk 4]
```diff
@@ -51,10 +51,7 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             return $data;
         }
 
-        if (
-            !($operation->canDeserialize() ?? true)
-            || !\in_array($method = $operation->getMethod(), ['POST', 'PUT', 'PATCH'], true)
```
Comment (vincentchalamon) [inline src/State/Provider/DeserializeProvider.php:56]: So it "_might_" be possible to deserialize a Get, GetCollection or Delete operations?
If we want to secure this behavior, maybe we should force `deserialize: false` in related operations

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: src/State/Provider/DeserializeProvider.php (hunk 4 above)
Comment (vincentchalamon) [reply src/State/Provider/DeserializeProvider.php:56]: Moved to MainController, see https://github.com/api-platform/core/pull/5857#discussion_r1341465506

**Review by vincentchalamon — APPROVED** [review]

**Pull request merged**